<launch>

  <node pkg="raspirover" type="raspirover_node" name="raspirover_node" output="screen" >
  </node>
  
  <node pkg="raspicam" type="raspicam_node" name="raspicam_node" >
    <param name="framerate" value="10"/>
    <param name="quality" value="50"/>
    <param name="width" value="320"/>
    <param name="height" value="240"/>
    <!--remap from="/camera/image/compressed" to="/image_raw/compressed"/-->
  </node>
  
  <node pkg="rtimulib_ros" type="rtimulib_ros" name="rtimulib_node" output="screen">
    <param name="calibration_file_path" value="$(find rtimulib_ros)/config"/>
    <param name="calibration_file_name" value="RTIMULib" />
    <param name="frame_id" value="imu_link"/>
    <remap from="/rtimulib_node/imu" to="/imu"/>
  </node>
 
  <include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch"/>  
  
  <node pkg="tf" type="static_transform_publisher" name="tf_00" args="0 0 0 0 0 0 /map /odom_combined 100"/>
  <node pkg="tf" type="static_transform_publisher" name="tf_02" args="0 0 0 0 0 0 /base_footprint /base_link 100"/>
  <node pkg="tf" type="static_transform_publisher" name="tf_03" args="0.04 0.02 0.04 0 0 0 /base_link /imu_link 100"/>
  <node pkg="tf" type="static_transform_publisher" name="tf_04" args="0.08 0.01 0.04 0 0 0 /base_link /sonar_link 100"/>
  <node pkg="tf" type="static_transform_publisher" name="tf_05" args="0.00 0.00 0.17 0 0 0 /base_link /laser_link 100"/>
  <node pkg="tf" type="static_transform_publisher" name="tf_06" args="0.02 0.0 0.08 0 0 0 /base_link /camera_link 100"/>
 
   <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping">
    <param name="scan" value="scan"/>
    <param name="map_update_interval" value="0.1"/>
    <param name="linearUpdate" value="0.1"/>
    <param name="angularUpdate" value="0.1"/>        
    <param name="maxUrange" value="1"/>
    <param name="maxRange" value="2"/>    
    <param name="iterations" value="1"/>
    <param name="delta" value="0.05"/>    
    <param name="particles" value="5"/>    
    <param name="minimumScore" value="200"/>
    <param name="odom_frame" value="odom_combined"/>
   </node> 

   <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf">
    <param name="output_frame" value="odom_combined"/>
    <param name="freq" value="10.0"/>
    <param name="sensor_timeout" value="1.0"/>
    <param name="odom_used" value="true"/>
    <param name="imu_used" value="true"/>
    <param name="vo_used" value="false"/>
    <param name="debug" value="false"/>
    <param name="self_diagnose" value="false"/>
    <remap from="/imu_data" to="/imu"/>
  </node>

  <!--
    <rosparam file="$(find chapter6_tutorials)/launch/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find chapter6_tutorials)/launch/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find chapter6_tutorials)/launch/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find chapter6_tutorials)/launch/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find chapter6_tutorials)/launch/base_local_planner_params.yaml" command="load" />
  -->
  
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <param name="controller_frequency" value="5"/>  
    <rosparam file="$(find raspirover)/config/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find raspirover)/config/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find raspirover)/config/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find raspirover)/config/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find raspirover)/config/base_local_planner_params.yaml" command="load" />
  </node>
 
</launch>